File: lock cracker.rb
def lock

    lock = "999"

    0.upto(9) do |i|
        0.upto(9) do |j|
            0.upto(9) do |k|

                i2 = i.to_s
                j2 = j.to_s
                k2 = k.to_s

                output = i2 + j2 + k2

                puts output

                if output == lock
                    puts "lock combination found: #{output}"
                    return
                end
                
            end
        end
    end
    puts "combination not found"
end

lock

File: lock cracker.rs
fn lock() {
    
    let lock = "583";

    for i in 0..10 {
        for j in 0..10 {
            for k in 0..10 {

                let i2 = i.to_string();
                let j2 = j.to_string();
                let k2 = k.to_string();

                let output = format!("{}{}{}", i2, j2, k2);

                println!("{}", output);

                if output == lock {
                    println!("lock combination found: {}", output);
                    return;
                }
            }
        }
    }
    println!("combination not found");
}

fn main() {
    lock();
}
